### PR TITLE
GraphNode ignore non-visible children for minimum size.

### PR DESCRIPTION
**Pull request by JohanAR**
Gives more flexibility when creating custom GraphNodes, by allowing the user to hide elements of choice when it gets small.

### CODE REVIEW

**Comment by YuriSizov** [comment]
While children's visible in tree status is cached, surely we only need their own visible status here? If the node itself, or its parents are invisible, this is irrelevant.

**Comment by JohanAR** [comment]
My use case is that I have a generic GraphNode subclass which has dynamically configured inputs and outputs. I solved this by hiding the children that aren't in use, and noticed that even the hidden nodes contributed to the GraphNode's minimum size. It also has a preview image, which never has left or right enabled in its slot, that I want to hide completely if the GraphNode is resized smaller than a certain size.

Perhaps it would be better to instantiate children to the GraphNode on demand instead of hiding the unused ones, but it felt logical to ignore hidden children when calculating min size.

**Review by Geometror — APPROVED** [review]
Seems reasonable to be, since `GraphNode` is a `Container` its behavior should be consistent with other containers. I couldn't spot any regressions in the VisualShader editor, but another opinion/some further testing would be good. There might be some hard to find cases where we expect the current behavior.

File: scene/gui/graph_node.cpp [hunk 1]
```diff
@@ -664,7 +664,7 @@ Size2 GraphNode::get_minimum_size() const {
 
 	for (int i = 0; i < get_child_count(); i++) {
 		Control *c = Object::cast_to<Control>(get_child(i));
-		if (!c) {
+		if (!c || !c->is_visible_in_tree()) {
```
Comment (YuriSizov) [inline scene/gui/graph_node.cpp:667]: What I was saying is that we don't need to check if it's visible in tree, we only need to check if it's visible itself.

```suggestion
		if (!c || !c->is_visible()) {
```

That's enough here, and would in fact be in line with what other containers are doing.

Comment (JohanAR) [reply scene/gui/graph_node.cpp]: Ah, sorry, I misunderstood. I have changed as you suggested and verified that this still works in my use case

**Review by YuriSizov — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.3.

**Pull request merged**